src/api/eth_filter.rs

```diff
@@ -103,6 +107,20 @@ impl<T: Transport, R> BaseFilter<T, R> {
     CallResult::new(self.transport.execute("eth_getFilterChanges", vec![id]))
   }
 
+  pub fn stream<'a>(&'a self, poll_interval: Duration) -> Box<Stream<Item = R, Error = Error> + 'a> where
```

Comment:
Wouldn't it be better to consume the filter? (then RwLocks in tests are not needed)

Also can we avoid `Boxing` the stream? By returning concrete struct that owns the filer we should be able to do that.